addons/mass_mailing/models/mail_notification.py

```diff
@@ -0,0 +1,124 @@
+    campaign_id = fields.Many2one('utm.campaign', related='mass_mailing_id.campaign_id', string='Campaign', index=True, store=True)
+    # Bounce and tracking
+    canceled = fields.Datetime(help='Date when the email has been invalidated. '
+                                    'Invalid emails are blacklisted, opted-out or invalid email format')
+    scheduled = fields.Datetime(help='Date when the email has been created', default=fields.Datetime.now)
+    sent = fields.Datetime(help='Date when the email has been sent')
+    exception = fields.Datetime(help='Date of technical error leading to the email not being sent')
+    opened = fields.Datetime(help='Date when the email has been opened the first time')
+    replied = fields.Datetime(help='Date when this email has been replied for the first time.')
+    bounced = fields.Datetime(help='Date when this email has bounced.')
+    # Link tracking
+    links_click_ids = fields.One2many('link.tracker.click', 'mailing_trace_id', string='Links click')
+    clicked = fields.Datetime(help='Date when customer clicked on at least one tracked link')
+    # Status
+    notification_status = fields.Selection(selection_add=[
+        ('outgoing', 'Outgoing'),
+        ('opened', 'Opened'),
+        ('replied', 'Replied')])
+    status_update = fields.Datetime('State Update', help='Last state update of the mail')
+
+    def name_get(self):
+        return [(n.id, '%s: %s (%s)' % (n.notification_type, n.mass_mailing_id.name or n.res_partner_id.display_name or '', n.id)) for n in self]
+
+    def _update_trace_status(self, values):
+        status_whitelist = ['sent', 'opened', 'clicked', 'replied', 'bounced', 'exception', 'canceled']
+        if any(key in status_whitelist for key in values.keys()):
+            notification_status = next((key for key in values.keys() if key in status_whitelist), False)
+            if notification_status == 'clicked':
+                notification_status = 'opened'
+            return notification_status
+
+    @api.model_create_multi
+    def create(self, values_list):
+        for values in values_list:
+            # Change default values if mass mailing is set
+            if 'mass_mailing_id' in values and values.get('mass_mailing_id'):
+                if 'notification_type' not in values:
+                    values['notification_type'] = 'email'
+                if 'mail_id' in values:
+                    values['mail_id_int'] = values['mail_id']
+                values['campaign_id'] = int(self.env['mailing.mailing'].browse(values.get('mass_mailing_id')).campaign_id)
+                status_info = self._update_trace_status(values)
+                if status_info:
+                    values['notification_status'] = status_info
+                    values['status_update'] = fields.Datetime.now()
+                else:
+                    values['notification_status'] = 'outgoing'
```

Comment:
And with that 'outgoing' default value you don't update the 'status_update' field? Seems strange.

I wonder if you could not default it to `fields.Datetime.now()` and avoid handling it in create.
Don't we always want to have a value for this?

Reply:
Mhh, well, It changes a little bit the functional usage, but I think it is more correct as you say. I am doing the change. Thanks